lib/CIndexStoreDB/CIndexStoreDB.cpp

```diff
@@ -94,25 +94,88 @@ class BlockIndexSystemDelegate: public IndexSystemDelegate {
+}
+
+void
+indexstoredb_creation_options_dispose(indexstoredb_creation_options_t c_options) {
+  auto *options = static_cast<CreationOptions *>(c_options);
+  delete options;
+}
+
+void
+indexstoredb_creation_options_add_prefix_mapping(indexstoredb_creation_options_t c_options,
+                                                 const char *path_prefix,
+                                                 const char *remapped_path_prefix) {
+  auto *options = static_cast<CreationOptions *>(c_options);
+  options->indexsStoreOptions.addPrefixMapping(path_prefix, remapped_path_prefix);
+}
+
+void
+indexstoredb_creation_options_set_listen_to_unit_events(indexstoredb_creation_options_t c_options,
+                                                        bool listenToUnitEvents) {
+  auto *options = static_cast<CreationOptions *>(c_options);
+  options->listenToUnitEvents = listenToUnitEvents;
+}
+
+void
+indexstoredb_creation_options_enable_out_of_date_file_watching(indexstoredb_creation_options_t c_options,
+                                                               bool enableOutOfDateFileWatching) {
+  auto *options = static_cast<CreationOptions *>(c_options);
+  options->enableOutOfDateFileWatching = enableOutOfDateFileWatching;
+}
+
+void
+indexstoredb_creation_options_set_readonly(indexstoredb_creation_options_t c_options,
+                                           bool readonly) {
+  auto *options = static_cast<CreationOptions *>(c_options);
+  options->readonly = readonly;
+}
+
+void
+indexstoredb_creation_options_set_wait(indexstoredb_creation_options_t c_options,
+                                       bool wait) {
+  auto *options = static_cast<CreationOptions *>(c_options);
+  options->wait = wait;
+}
+
+void
+indexstoredb_creation_options_set_use_explicit_output_units(indexstoredb_creation_options_t c_options,
+                                                            bool useExplicitOutputUnits) {
+  auto *options = static_cast<CreationOptions *>(c_options);
+  options->useExplicitOutputUnits = useExplicitOutputUnits;
+}
+
 indexstoredb_index_t
 indexstoredb_index_create(const char *storePath, const char *databasePath,
                           indexstore_library_provider_t libProvider,
                           indexstoredb_delegate_event_receiver_t delegateCallback,
-                          bool useExplicitOutputUnits, bool wait, bool readonly,
-                          bool enableOutOfDateFileWatching, bool listenToUnitEvents,
+                          indexstoredb_creation_options_t cOptions,
                           indexstoredb_error_t *error) {
 
   auto delegate = std::make_shared<BlockIndexSystemDelegate>(delegateCallback);
   auto libProviderObj = std::make_shared<BlockIndexStoreLibraryProvider>(libProvider);
+  auto options = static_cast<CreationOptions *>(cOptions);
 
   std::string errMsg;
   if (auto index =
           IndexSystem::create(storePath, databasePath, libProviderObj, delegate,
```

Comment:
@bnbarham Is this alright or should the `CreationOptions` be passed further down instead?

Reply:
Done